src/role.rs

```diff
@@ -10,22 +11,35 @@ pub(crate) enum Role {
     Client { rt: Rt, handle: JoinHandle<()> },
 
     /// A simulated host
-    Simulated(Rt),
+    Simulated {
+        rt: Rt,
+        software: Box<dyn Fn() -> Pin<Box<dyn Future<Output = ()> + 'static>>>,
```

Comment:
How about some crate types for Software/Factory?

Reply:
I'll defer this until I find myself repeating the signature. Otherwise it's just indirection for no benefit. The purpose is already named (by the field).

Reply:
I added it as there actually *was* repetition due to type inference not working.